app/ui/shared/widgets/autocompleted-search.jsx

```diff
@@ -35,13 +35,30 @@ class AutocompletedSearch extends Component {
   constructor(props) {
     super(props);
     this.shouldComponentUpdate = PureRenderMixin.shouldComponentUpdate.bind(this);
+    this.handleWindowClick = this.handleWindowClick.bind(this);
 
     this.state = {
       showSelectionList: false,
       selectedIndex: 0,
     };
   }
 
+  componentWillMount() {
+    if (typeof window === 'object') {
+      window.addEventListener('click', this.handleWindowClick);
+    }
+  }
+
+  componentWillUnmount() {
+    if (typeof window === 'object') {
```

Comment:
@victorporof  I'm not sure what you mean by this -- a window safety? or unmount clean up? The closing on escape/click seems to be working fine for it. Can you clarify in a follow up?